contracts/token/InvestorVesting.sol

```diff
@@ -0,0 +1,149 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.1;
+
+import "./SyndicateERC20.sol";
+import "../utils/Ownable.sol";
+
+import "hardhat/console.sol";
+
+contract InvestorVesting is Ownable {
+  event TGETriggered(uint256 nowTimestamp, uint256 timestampTGE);
+
+  address public syn;
+  uint256 public timestampTGE;
+  uint256 private _previouslyInvested;
+
+  struct VestingSchedule {
+    uint8 tge;
+    uint8 firstWeek;
+    uint8 secondWeek;
+    uint8 thirdWeek;
+    uint8 fourthWeek;
+    uint8 cliff;
+    uint8 remainingMonths;
+  }
+
+  mapping(uint8 => VestingSchedule) public vestingSchedules;
+
+  struct Investment {
+    uint8 tier;
+    uint120 amount;
+    uint120 claimed;
+  }
+
+  mapping(address => Investment) public investments;
+
+  constructor(address _syn) {
+    syn = _syn;
+    // seed
+    vestingSchedules[1] = VestingSchedule({
+      tge: 1,
+      firstWeek: 2,
+      secondWeek: 3,
+      thirdWeek: 4,
+      fourthWeek: 5,
+      cliff: 12,
+      remainingMonths: 36
+    });
+    // strategic
+    vestingSchedules[2] = VestingSchedule({
+      tge: 1,
+      firstWeek: 2,
+      secondWeek: 3,
+      thirdWeek: 5,
+      fourthWeek: 7,
+      cliff: 12,
+      remainingMonths: 36
+    });
+    //
+    vestingSchedules[3] = VestingSchedule({
+      tge: 2,
+      firstWeek: 2,
+      secondWeek: 2,
+      thirdWeek: 2,
+      fourthWeek: 2,
+      cliff: 10,
+      remainingMonths: 36
+    });
+    //
+    vestingSchedules[4] = VestingSchedule({
+      tge: 2,
+      firstWeek: 4,
+      secondWeek: 6,
+      thirdWeek: 9,
+      fourthWeek: 12,
+      cliff: 9,
+      remainingMonths: 36
+    });
+  }
+
+  function triggerTGE(uint256 _tgeTimestamp) external onlyOwner {
+    if (_tgeTimestamp == 0) {
+      _tgeTimestamp = block.timestamp;
+    }
+    require(_tgeTimestamp >= block.timestamp, "InvestorVesting: cannot set up TGE before now");
+    require(_tgeTimestamp < block.timestamp + 1 weeks, "InvestorVesting: cannot set up TGE over a week from now");
+    timestampTGE = _tgeTimestamp;
+    emit TGETriggered(block.timestamp, timestampTGE);
+  }
+
+  // must be called many times to address all the investors by tier
+  function init(
+    uint8 _tier,
+    address[] memory _investors,
+    uint120[] memory _amounts
+  ) external onlyOwner {
+    require(timestampTGE == 0, "InvestorVesting: TGE already triggered");
+    require(_amounts.length == _investors.length, "InvestorVesting: lengths do not match");
+    require(_tier >= 1 && _tier <= 4, "InvestorVesting: tier not found");
+    uint256 totalInvestments = 0;
+    for (uint256 i = 0; i < _investors.length; i++) {
+      // do not require to avoid reverting
+      if (investments[_investors[i]].tier == 0) {
```

Comment:
so it only processes tier 0?  how about the rest?

Reply:
That is an initial setting. If tier is != zero it means that it has already set.